habitat_baselines/common/obs_transformers.py

```diff
@@ -217,113 +218,496 @@ def from_config(cls, config: Config):
+        mapx = 2 * u / self.img_w - 1.0
+        mapy = 2 * v / self.img_h - 1.0
+        proj_pts = torch.stack([mapx, mapy], dim=-1)
+
+        # Valid mask
+        valid_mask = torch.abs(proj_pts).max(-1)[0] <= 1  # -1 <= grid.xy <= 1
+        valid_mask *= img_pts[..., 2] > 0
+        return proj_pts, valid_mask
+
+    def unprojection(
+        self, with_rotation: bool = True
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        v, u = torch.meshgrid(
+            torch.arange(self.img_h), torch.arange(self.img_w)
+        )
+        x = (u + 0.5) - self.img_w / 2
+        y = (v + 0.5) - self.img_h / 2
+        z = torch.full_like(x, self.f, dtype=torch.float)
+        unproj_pts = torch.stack([x, y, z], dim=-1)
+        # Project on unit shpere
+        unproj_pts /= torch.norm(unproj_pts, dim=-1, keepdim=True)
+        # All points in image are valid
+        valid_mask = torch.full(unproj_pts.shape[:2], True, dtype=torch.bool)
+
+        # Rotate unproj_pts points according to camera rotation
+        if with_rotation:
+            unproj_pts = self.camcoord2worldcoord(unproj_pts)
+
+        return unproj_pts, valid_mask
+
+
+class EquirecCamera(CameraProjection):
+    """This is the equirectanglar camera class."""
+
+    def __init__(
+        self, img_h: int, img_w: int, R: Optional[torch.Tensor] = None
+    ):
+        """Args:
+        img_h: (int) the height of equirectanglar camera image
+        img_w: (int) the width of equirectanglar camera image
+        R: (torch.Tensor) 3x3 rotation matrix of camera
+        """
+        super(EquirecCamera, self).__init__(img_h, img_w, R)
+
+    def projection(
+        self, world_pts: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        # Rotate world points according to camera rotation
+        world_pts = self.worldcoord2camcoord(world_pts)
+
+        x, y, z = world_pts[..., 0], world_pts[..., 1], world_pts[..., 2]
+        # x,y,z to theta, phi
+        theta = torch.atan2(x, z)
+        c = torch.sqrt(x * x + z * z)
+        phi = torch.atan2(y, c)
+
+        # For grid_sample, -1 <= proj_pts <= 1
+        mapx = theta / np.pi
+        mapy = phi / (np.pi / 2)
+        proj_pts = torch.stack([mapx, mapy], dim=-1)
+
+        # All points in image are valid
+        valid_mask = torch.full(proj_pts.shape[:2], True, dtype=torch.bool)
+        return proj_pts, valid_mask
+
+    def unprojection(
+        self, with_rotation: bool = True
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        theta_map, phi_map = self.get_theta_phi_map(self.img_h, self.img_w)
+        unproj_pts = self.angle2sphere(theta_map, phi_map)
+        # All points in image are valid
+        valid_mask = torch.full(unproj_pts.shape[:2], True, dtype=torch.bool)
+        # Rotate unproj_pts points according to camera rotation
+        if with_rotation:
+            unproj_pts = self.camcoord2worldcoord(unproj_pts)
+        return unproj_pts, valid_mask
+
+    # Get theta and phi map
```

Comment:
Just make these comments into actual doc strings, because why not. That way they should up on autocomplete tools for IDEs

Reply:
Thanks! I didn't think of it. I modified it so that it will appear in IDEs.